doc/rtd/topics/faq.rst

```diff
@@ -166,12 +166,27 @@ To wait for clous-init to complete, run:
 
         $ cloud-init status --wait
 
+Failing to Complete on Systemd:
+-------------------------------
+
+Cloud-init consists of multiple services on systemd. If a service
+that cloud-init depends on stalls, cloud-init will not continue.
+If reporting a bug related to cloud-init failing to complete on
+systemd, please make sure to include the following logs.
+
+.. code-block:: shell-session
+
+        $ systemd-analyze critical-chain cloud-init.target
```

Comment:
Not sure if this would add a lot of info but maybe is a good thing:

```suggestion
        $ systemd-analyze critical-chain cloud-init.target
        $ journalctl --boot=-1
```

Reply:
Good idea. On long-running systems with limited storage there is a chance that logs from former boots will be garbage collected by systemd, but worth suggesting anyhow I think.